pkg/pfconfig/shared/shared.go

```diff
@@ -33,7 +33,7 @@ import (
 // SX1301Config contains the configuration for the SX1301 concentrator.
 type SX1301Config struct {
 	LoRaWANPublic       bool
-	ClockSource         uint8
+	ClockSource         uint8 `json:"omitempty"`
```

Comment:
1. Your `json` tag now renames this field to `omitempty`
2. This has no effect, because `SX1301Config` has a `MarshalJSON` implementation. That's where you need to change this.
3. You need to make this a `*uint8` so that we can make a distinction between 0 and "omit"